app/services/oauth_clients/create_service.rb

```diff
@@ -35,5 +35,12 @@
 # The comments here are also valid for the other *_service.rb files
 module OAuthClients
   class CreateService < ::BaseServices::Create
+    protected
+
+    def before_perform(params, _service_result)
+      OAuthClient.where(integration: params[:integration]).delete_all
+
+      super
```

Comment:
Please extend the spec for this service to reflect this deviation from the default behaviour.

Reply:
amended spec